tensorbay/utility/attr.py

```diff
@@ -0,0 +1,220 @@
+        """
+        contents: Dict[str, Any] = {}
+        for support in getattr(self, "_supports", []):
+            contents.update(support._dumps(self))  # pylint: disable=protected-access
+
+        for attribute, field_object in self._attrs_fields.items():
+            content_key = field_object.content_key if field_object.content_key else attribute
+            if hasattr(self, attribute):
+                contents[content_key] = field_object.dumps(getattr(self, attribute))  # type: ignore
+                continue
+
+            if field_object.default:
+                contents[content_key] = field_object.default
+
+        return contents
+
+
+def field(
+    *,
+    is_dynamic: bool = False,
+    content_key: Optional[str] = None,
+    default: Optional[Any] = None,
+    error_message: Optional[str] = None,
+) -> Any:
+    """Return an object to identify dynamic_attr fields.
+
+    Arguments:
+        is_dynamic: Determine if this is a dynamic attribute.
+        content_key: Display value of the attribute.
+        default: Default value of the attribute.
+        error_message: The custom error message of the attribute.
+
+    Raises:
+        ValueError: Specify both is_dynamic=False and default value.
+
+    Returns:
+        A Field object containing all arguments.
+    """
+    if not is_dynamic and default:
+        raise ValueError("Cannot specify both is_dynamic=False and default value")
+    return Field(is_dynamic, content_key, default, error_message)
+
+
+def _get_field(obj: Any, name: str, annotation: Any) -> Field:
+
+    default = getattr(obj, name, None)
+    if isinstance(default, Field):
+        field_object = default
+    else:
+        field_object = Field()
+
+    field_object.type = annotation
+    field_object.loads, field_object.dumps = _get_operate_type(annotation)
+
+    return field_object
+
+
+def _get_operate_type(
+    annotation: Any, is_internal: bool = False
+) -> Tuple[Callable[[Any], Any], Callable[[Any], Any]]:
+    """Convert attribute operate method from annotations.
+
+     AttrMixin has three operate types which is classified by attribute annotation.
+        1. builtins operate, like str, int
+        2. tensorbay custom class, like tensorbay.label.Classification
+        3. tensorbay custom class list, like [tensorbay.label.LabeledBox2D]
+
+    Arguments:
+        annotation: Type of the attribute.
+        is_internal: Whether the annotation is inside of typing.List.
+
+    Returns:
+        operate methods of the annotation
+    """
+    if str(annotation).startswith("typing.List"):
+        return _get_operate_type(annotation.__args__[0], True)
+
+    if getattr(annotation, "__module__", None) in {"builtins", None}:
+        return (lambda contents: contents, lambda attribute: attribute)
+
+    if not is_internal:
+        return (
+            lambda contents: annotation.loads(contents),  # pylint: disable=unnecessary-lambda
+            lambda attribute: attribute.dumps(),
```

Comment:
better to not use naming `attribute`, that will make `attribute` ambiguous in our SDK.